staging/src/k8s.io/kubectl/pkg/cmd/util/factory.go

```diff
@@ -61,6 +62,8 @@ type Factory interface {
 
 	// Returns a schema that can validate objects stored on disk.
 	Validator(validate bool) (validation.Schema, error)
-	// OpenAPISchema returns the schema openapi schema definition
+	// OpenAPISchema returns the parsed openapi schema definition
 	OpenAPISchema() (openapi.Resources, error)
+	// OpenAPIGetter returns a getter for the openapi schema document
+	OpenAPIGetter() discovery.OpenAPISchemaInterface
```

Comment:
So my only objection is why would I care about these 2 separately instead of just having a single method `OpenAPISchema` do the heavy lifting for me like it did before with just the caching added?

Reply:
Because unfortunately we have two different types: `openapi.Resources` which is our own format for the openapi, and the actual openapi. Some operations require the former, some operations require the latter 😞

Reply:
I spoke with Antoine, it seems that the disconnection between `openapi.Resource` and `openapi_v2.Document` is too big to be addressed at this time. We agree that we'll revisit that at a later time. This is good to go as soon as it gets green.